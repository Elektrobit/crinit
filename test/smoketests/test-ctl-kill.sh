# shellcheck shell=bash
# SPDX-License-Identifier: MIT
#
# smoketest for crinit-ctl kill command
#

setup() {
    crinit_config_setup
}

run() {
    crinit_daemon_start "${SMOKETESTS_CONFDIR}"/demo.series
    sleep 3

    if ! "${BINDIR}"/crinit-ctl addtask "${SMOKETESTS_CONFDIR}"/sleep_one_day.crinit; then
        echo "crinit-ctl addtask failed"
        return 1
    fi

    sleep 3

    if ! "$BINDIR"/crinit-ctl kill sleep_one_day; then
        echo "crinit-ctl kill failed"
        return 1
    fi

    sleep 1

    if ! crinit_task_check_status "sleep_one_day" "failed"; then
        return 1
    fi
}

teardown() {
    # Terminate crinit daemon
    crinit_daemon_stop
}
