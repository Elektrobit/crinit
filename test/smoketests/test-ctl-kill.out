Status: failed, PID: -1
